vsphere/internal/virtualdevice/virtual_machine_cdrom_subresource.go

```diff
@@ -472,27 +508,39 @@ func (r *CdromSubresource) Update(l object.VirtualDeviceList) ([]types.BaseVirtu
 		return nil, fmt.Errorf("device at %q is not a virtual CDROM device", l.Name(d))
 	}
 
-	// To update, we just re-insert the ISO as per create, and send it as an edit.
 	dsID := r.Get("datastore_id").(string)
 	path := r.Get("path").(string)
-	ds, err := datastore.FromID(r.client, dsID)
+	clientDevice := r.Get("client_device").(bool)
+	err = r.ValidateCdromConfig()
 	if err != nil {
-		return nil, fmt.Errorf("cannot find datastore: %s", err)
-	}
-	dsProps, err := datastore.Properties(ds)
-	if err != nil {
-		return nil, fmt.Errorf("could not get properties for datastore: %s", err)
-	}
-	dsName := dsProps.Name
-
-	dsPath := &object.DatastorePath{
-		Datastore: dsName,
-		Path:      path,
+		return nil, err
 	}
 
-	device = l.InsertIso(device, dsPath.String())
-	l.Connect(device)
+	if dsID != "" && path != "" {
```

Comment:
This might be a good candidate for a unexported helper as this code looks pretty much the same as it does in `Create`.